eth/db/journal.py

```diff
@@ -290,9 +305,13 @@ def has_clear(self) -> bool:
         return self.journal.has_clear(self.journal.root_changeset_id)
 
     def __delitem__(self, key: bytes) -> None:
-        if key not in self.journal and key not in self.wrapped_db:
-            raise KeyError(key, "key could not be deleted in JournalDB, because it was missing")
-        del self.journal[key]
+        if key in self.wrapped_db:
```

Comment:
Do you know if any callers skip the pipeline and talk to `wrapped_db` directly? `JournalDB` assumes not, and this could lead to some weird bugs. If a `ERASE` is pushed but the database later has the value then it won't be deleted, if a `DELETE` is pushed but the database later loses the value `persis()` will fail, etc.

Reply:
Yeah, I think this is broadly true of our database chains (for example, `CacheDB` will be confused if something it wraps is changed out-of-band). I think this is just something we live with in the current model of wrapping. It's the pipeliner's responsibility to not access the wrapped DB, or to have an understanding of how to fix it up if they do. (eg~ by calling `reset_cache()` on `CacheDB`)